wallet/addresses.go

```diff
@@ -977,6 +977,49 @@ func (w *Wallet) AccountBranchAddressRange(account, branch, start, end uint32) (
 	return addrs, nil
 }
 
+// ImportedAddresses returns each of the addresses imported into an account.
+func (w *Wallet) ImportedAddresses(ctx context.Context, account string) (_ []KnownAddress, err error) {
+	const opf = "wallet.ImportedAddresses(%q)"
+	defer func() {
+		if err != nil {
+			op := errors.Opf(opf, account)
+			err = errors.E(op, err)
+		}
+	}()
+
+	if account != "imported" {
+		return nil, errors.E("account does not record imported keys")
+	}
+
+	var addrs []KnownAddress
+	err = walletdb.View(ctx, w.db, func(dbtx walletdb.ReadTx) error {
+		ns := dbtx.ReadBucket(waddrmgrNamespaceKey)
+		f := func(a udb.ManagedAddress) error {
+			ma := managedAddress{
+				acct:     account,
+				acctKind: AccountKindImported,
+				addr:     a,
+			}
+			switch a.(type) {
+			case udb.ManagedPubKeyAddress:
+				ma.script = ma.p2pkhScript
```

Comment:
Shouldn't this be getting the script from `a`?  `ma.p2pkhScript` will be nil here.

Reply:
No, this is correct.  `script` is a struct member of type `func() (uint16, []byte)` and `p2pkhScript` and `p2shScript` are methods which are assigned to the field.

Reply:
Ah, I hadn't updated to the latest code which made `script` a func instead of a `[]byte` as the name would imply.  Thanks for the clarification.